packages/eslint-plugin/docs/rules/prefer-regexp-exec.md

```diff
@@ -0,0 +1,38 @@
+# Enforce to use `RegExp#exec` over `String#match` (prefer-regexp-exec)
+
+`String#match` is outperformed by `RegExp#exec` and both work in the same way, except for /g flag.
+
+## Rule Details
+
+This rule is aimed at enforcing the more performant way for applying regular expressions on Strings.
+
+Examples of **incorrect** code for this rule:
+
+```ts
+'something'.match(/thing/);
+
+const text = 'something';
+const search = /thing/;
```

Comment:
It's inconsistent that you use variables here, but not in the "correct" example.